### DISCUSSION TITLE
Implementing Github actions

### DISCUSSION THREAD

**Discussion by Philbywhizz** [togfoxy/MarsLander#168] [General]
Hi Team,

I'm currently playing around with github actions and intergrating luacheck into the github pipeline to see how it can help us.

It shouldn't affect you in any way, this is more of a FYI.

**Comment by togfoxy** [comment]
I use only the web page github. You think that will help me as well? #curious

**Reply by Philbywhizz** [reply]
Well the idea is that when you submit a pull request it runs luacheck against the code.

That's the idea.

**Comment by Philbywhizz** [comment]
I've hooked up the 'luacheck' github action to pull request and pushes to main.

What does that mean? It means that luacheck will run in the background and verify the code. It will give either a green tick or a red tick for the status of the repository for luacheck.

Right now though it is showing a red cross, because there are lots of warnings. We can either turn the warnings off or we can fix the code.

It won't prevent you from merging or accepting pull requests, just doesn't make the button green by default.

Does that make sense to everyone? Github actions are fun once you get the hang of them. :)

**Reply by MadByteDE** [reply]
is it possible to add another status to show that warnings are present? Fail sounds so harsh for something like trailing whitespaces :smile: 
If not I'd say it's fine as is.

**Reply by Philbywhizz** [reply]
Let me see if I can change it to only error when there is an error (warnings ignored)

**Comment by togfoxy** [comment]
It is throwing a lot of errors on the enums. Will take time to get use to it.

A step forward none-the-less!

**Reply by MadByteDE** [reply]
I think with more refactoring, this issue will be resolved :wink:

**Comment by Philbywhizz** [comment]
I've set the github action to only throw an error when there are syntax errors.

So as of right now, if you check in code that has a syntax error detected by luacheck then it will flag it as an error. The warnings are supressed for now. I think once we sort through the warnings we can perhaps enable it at a later date.

**Comment by togfoxy** [comment]
Remind me again how I get those warnings? I think you made a comment on a branch that is now deleted.

**Reply by MadByteDE** [reply]
Download and run luacheck locally on the source directory.

**Reply by Philbywhizz** [reply]
For [reference](https://github.com/togfoxy/MarsLander/pull/180#issuecomment-962537327). I just closed/cancelled the pull request, not deleted. All the comments are still there.
